lib/Server.js

```diff
@@ -3326,6 +3326,12 @@ class Server {
     if (typeof this.options.onListening === "function") {
       this.options.onListening(this);
     }
+
+    if (this.options.hot) {
+      // TODO fix me in the next major release
+      // @ts-ignore
+      process.env.WEBPACK_HOT = 'true';
+    }
```

Comment:
It can be too late here, do you test it locally?

Reply:
Hmm, I think we should move this to `normalizeOptions()`
https://github.com/webpack/webpack-dev-server/blob/8817fa88edbe11259375a889cbabc4c7d0c603ac/lib/Server.js#L799

Reply:
Ideally we should set it in webpack-cli because it is an enviroment variable and we have to set it before we loading webpack configuration, so let's close it here and implement this on webpack-cli level, because I think we will not remove `--hot` never we can hardcode such logic

Reply:
But we can also use webpack-dev-server via the Nodejs API that doesn't require webpack-cli. In that case this variabke will be missing.

Reply:
But we can also use webpack-dev-server via the Nodejs API that doesn't require webpack-cli. In that case this variable will be missing.

Reply:
But we still run webpack-cli https://github.com/webpack/webpack-dev-server/blob/master/bin/webpack-dev-server.js#L103 :smile:

Reply:
Yeah, but in this case it is not our problem to set env variables (because when you have custom cli runner you can set any own variables without problem because I control own code), for webpack-cli we just have couple of built-in env variables and not more - popular cases